src/character.cpp

```diff
@@ -7961,6 +7969,16 @@ ret_val<void> Character::can_wield( const item &it ) const
         mount->type->mech_weapon && it.typeId() != mount->type->mech_weapon ) {
         return ret_val<void>::make_failure( _( "You cannot wield anything while piloting a mech." ) );
     }
+    if( has_trait( trait_VEGAN ) && ( it.made_of( material_leather ) ||
+                                      it.has_own_flag( flag_ANIMAL_PRODUCT ) ||
+                                      it.made_of( material_fur ) ||
+                                      it.made_of( material_wool ) ||
+                                      it.made_of( material_chitin ) ||
+                                      it.made_of( material_bone ) ||
+                                      it.made_of( material_gutskin ) ||
+                                      it.made_of( material_acidchitin ) ) ) {
```

Comment:
If we are going down this route, then this selection looks very similar to the existing selection in `Character::can_wear`: https://github.com/CleverRaven/Cataclysm-DDA/blob/4bf90cd464f6d099004e98bd24be4aed1d931a82/src/character_attire.cpp#L129-L136

Is it possible to make a reusable method for this to capture the intent on both code places? Something like a `despised_by_vegans` method?